pinot-minion/src/main/java/org/apache/pinot/minion/api/resources/PinotTaskProgressResource.java

```diff
@@ -84,4 +86,34 @@ public String getSubtaskProgress(
           .build());
     }
   }
+
+  @GET
+  @Path("/tasks/subtask/state/{subTaskState}/progress")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation("Get finer grained task progress tracked in memory for subtasks with the given state")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Internal server error")
+  })
+  public String getSubtaskWithGivenStateProgress(
```

Comment:
I think that we can merge this function with `/tasks/subtask/progress`?

Reply:
There is a minor problem when we merge those 2 methods.

The existing `/tasks/subtask/progress` returns response in format of 
`{"taskname":["....", "....", "..."]}`; 
the newly-added one returns response in format of `{"taskname":{"progress":["....", "....", "..."], "taskState":"", "startTs":XXXX}}`

I am concerned about (1) returning two different formats in the same method, or (2) changing the return format of the existing public API.

Any suggestions?

Reply:
I see. Let's add the second function but let's make it work for both. We can slowly deprecate the existing one favoring the new API.